Dev/add spdx2 import agent

maxhbr opened the pull request:
this is WIP

maxhbr committed:
feat(spdx2Import): inital commit

- implement agent and ui
- adopt license-explorer and others
- allow to download the report which was uploaded by the user
- add new dependency (EasyRdf) to composer

maxhbr committed:
feat(spdx2Import): also import copyright statements

* they get saved in the DB and also exported
* they are __not__ yet displayed by the copyright ui

maxhbr committed:
feat(spdx2Import): conclusions

maxhbr committed:
feat(spdx2Import): start to make conclusions optional

maxhbr committed:
feat(spdx2Import): compare only by sha1

maxhbr committed:
feat(spdx2Import): menu entry at "Upload::..."

deveaud-m committed:
feat(obligations): extend datamodel and obligation management

* ob_type: type of obligation {Obligation, Restriction, Risk or Right}
* ob_classification: the level of attention the obligation could become (optional)
* ob_modifications: does this obligation concerns modified source code (optional)
* ob_comment: from the obligation admin (optional)

Support the association of reference licenses and/or
candidate licenses for each obligation. CSV import and
export functions extended as well, one new column was
added in the resulting CSV table.

Additionaly: add the ability to delete an obligation and
throws a confirm dialog box.

max-wittig committed:
fix(debian9): add compatibility with debian9

bill-auger committed:
fix(install): update packages deps for latest debian and ubuntu

 * debian stretch,buster,sid
 * ubuntu yakkety,zesty,artful

maximilianhuber commented:
current problem after rebase:
```
[...]
fossology-scheduler_1  | 2017-07-11 11:19:28 scheduler [1] :: JOB[-19].spdx2Import[70.localhost]: agent didn't send version information: "PHP Parse error:  syntax error, unexpected '$this' (T_VARIABLE) in /usr/local/share/fossology/spdx2Import/agent/spdx2Import.php on line 388"
fossology-scheduler_1  | 2017-07-11 11:19:28 scheduler [1] :: JOB[-19].spdx2Import[70.localhost]: agent was killed by signal: 9.Killed
fossology-scheduler_1  | 2017-07-11 11:19:28 scheduler [1] :: WARNING agent.c.959: JOB[-19].spdx2Import[70.localhost]: agent closed unexpectedly, agent status was AGENT_FAILED
fossology-scheduler_1  | 2017-07-11 11:19:28 scheduler [1] :: ERROR agent.c.970: agent localhost.spdx2Import has failed scheduler startup test
```

maximilianhuber commented:
TODO:
- [ ] different scanner findings for `licenseInfoInFile` and `concludedLicenses`
- [ ] move the conclusion-action to the decider
- [ ] set main license
  - is located below clearing result?

maximilianhuber commented:
corresponding public issue https://github.com/fossology/fossology/issues/669

maxhbr committed:
feat(spdx2Import): refactoring and splitup of files

maximilianhuber commented:
Minimal setup for easyrdf testing:
```diff
diff --git a/composer.json b/composer.json
new file mode 100644
index 0000000..710c866
--- /dev/null
+++ b/composer.json
@@ -0,0 +1,7 @@
+{
+    "name": "minimal easyrdf",
+    "require": {
+        "php": ">=5.3.0",
+        "easyrdf/easyrdf": "0.9.*"
+    }
+}
diff --git a/public/index.php b/public/index.php
new file mode 100644
index 0000000..682b28b
--- /dev/null
+++ b/public/index.php
@@ -0,0 +1,17 @@
+<pre>
+<?php
+
+require '../vendor/autoload.php';
+
+use EasyRdf_Graph;
+
+$filename = "./SPDXRdfExample.rdf";
+
+/** @var EasyRdf_Graph */
+$graph = new EasyRdf_Graph();
+$graph->parseFile($filename, 'rdfxml', null);
+$index = $graph->toRdfPhp();
+
+var_dump($index);
+?>
+</pre>
\ No newline at end of file
```

maximilianhuber commented:
@mcjaeger The UI currently looks like:
![2017-07-20_17 36 01](https://user-images.githubusercontent.com/1187050/28425940-0351ca40-6d72-11e7-9fae-4fe30a3d2440.png)

maximilianhuber commented:
Links:
- http://www.easyrdf.org/docs/api/EasyRdf_Graph.html
- http://www.easyrdf.org/docs/api/EasyRdf_Resource.html

deveaud-m committed:
fix(obligations): correct php syntax using phpcs

maxhbr committed:
feat(spdx2Import): splitup to support other formats

maxhbr committed:
refactor(reportImport): spdx2Import -> reportImport

maxhbr committed:
feat(reportImport): handle all arguments from UI

maxhbr committed:
feat(reportImport): add option to create real licenses

maxhbr committed:
feat(spdx2): also export ninka and import data

maxhbr committed:
feat(reportImport): parse also xml files

maxhbr committed:
feat(reportImport): minor improvements

maxhbr committed:
feat(docker-compose): increase apache verbosity

maxhbr committed:
feat(reportImport): minor changes to satisfy older PHP versions

maxhbr committed:
feat(reportImport): add corresponding debain definitions

deveaud-m committed:
fix(obligations): rename obligation to license map table

This ensures compatibility with running instance from
FOSSology containing the previous version of the
obligation feature, the table mapping obligations to license
references was renamed back from `obligation_license_map` to
`obligation_map`.

maxhbr committed:
fix(reportImport): fix bug in reportImport, refactor file matching

max-wittig committed:
chore(changelog): update to commitlint

deveaud-m committed:
fix(obligations): select only single value for ob_classification and ob_type

maxhbr committed:
feat(spdx2): bump output version from 2.0 to 2.1

fixes: #902
closes: #899

maxhbr committed:
feat(copyright): allow to have multiple copyright decisions

related to #906 and #882

maxhbr committed:
Merge pull request #897 from max-wittig/chore/changelog

chore(changelog): update to commitlint

tested-by: maximilian.huber@tngtech.com
review-by: maximilian.huber@tngtech.com

mcjaeger committed:
Merge pull request #892 from maxhbr/dev/DockerIncreaseApacheVerbosity

feat(docker-compose): increase apache verbosity

review-b:michael.c.jaeger@siemens.com

mcjaeger committed:
Merge pull request #867 from fossology/improved-obligations

feat(obligations): extend database and obligation management with new data:

review-by:max.wittig@siemens.com
tested-by:max.wittig@siemens.com,michael.c.jaeger@siemens.com

maxhbr committed:
chore(gitignore): add more entries to the blacklist

maxhbr committed:
feat(reportImport): add imported coyprights as decisions

mcjaeger committed:
Merge pull request #857 from siemens/fix/debian

fix(debian9): add compatibility with debian9


review-by:maximilian.huber@tngtech.com
tested-by:michael.c.jaeger@siemens.com

mcjaeger committed:
Merge pull request #903 from maxhbr/dev/bumpSpdxOutputVersion

faet(spdx2): bump output version from 2.0 to 2.1

tested-by:michael.c.jaeger@siemens.com

maxhbr committed:
feat(reportImport): handle `orLaterOperator` correctly

maxhbr committed:
feat(reportImport): some cleanup and minor improvements
